stdlib/strtod.c

```diff
@@ -90,34 +115,154 @@ double strtod(const char *__restrict nptr, char **__restrict endptr)
 	}
 
 	res = (double)integer;
-	for (i = 0; i < 9; i++) {
-		if ((overflow_int >> i) & 1)
+	for (i = 0; i < RANGE_OF_EXPONENT_DOUBLE_64; i++) {
+		if ((overflow_int >> i) & 1) {
 			res *= exp_val;
+		}
+
 		exp_val *= exp_val;
 	}
 
 	res += (double)frac * pow(10, -fracsz);
 	exp_val = 10;
-	for (i = 0; i < 9; i++) {
+	for (i = 0; i < RANGE_OF_EXPONENT_DOUBLE_64; i++) {
 		if ((exp >> i) & 1) {
-			if (exp_sign == -1)
+			if (exp_sign == -1) {
 				res /= exp_val;
-			else
+			}
+			else {
 				res *= exp_val;
+			}
 		}
+
 		exp_val *= exp_val;
 	}
 
-	if (sign)
+	if (sign) {
 		res = -res;
+	}
 
-	if (isnum && endptr != NULL)
+	if (isnum && endptr != NULL) {
 		*endptr = (char *)nptr;
+	}
 
 	return isnum ? res : 0;
 }
 
 
+long double strtold(const char *__restrict nptr, char **__restrict endptr)
```

Comment:
shame it's almost ^C^V of the `strtod` but I don't have a clear solution on how to unify the implementations.

Was this implementation tested?